hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java

```diff
@@ -179,7 +171,8 @@ public int getId() {
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeUTF(groupName);
-        out.writeUTF(groupPassword);
+        //TODO @tkountis remove in 4.0
+        out.writeUTF(EMPTY_PWD);
```

Comment:
All accurate points, thank you. However, I see a downside, which is major, security wise. The most often use-case of the ConfigCheck is during the 1st minute of a Node going up, which sends rapidly datagrams, until it discovers a cluster. During that period, single node (never joined) no-matter what state the cluster is at, we leak the password with this approach.

So imagine the above scenario with a 3.8 EE cluster, which we rolling-upgrade to 3.9 EE, one by one, successfully, so now the occasional ConfigCheck payload, from the cluster to the 'wild', is properly skipping the password, but now we expand the cluster by adding a new node, which now again leaks the password, until it joins.

Of course I am ok with both solutions, but in your approach we will have to go to 4.0 totally stop leaking the pwd, whereas in my approach we can do it now, with the compromise that the client needs to first upgrade to latest 3.8 patch, which understandably is a burden. 

I guess this is a TDD decision, so do you mind adding your comment in the TDD ?